autoload/ctrlp.vim

```diff
@@ -2735,8 +2742,29 @@ fu! ctrlp#init(type, ...)
 	en
 	cal s:BuildPrompt(1)
 	if s:keyloop | cal s:KeyLoop() | en
-	return 1
+	retu 1
+endf
+
+" - Events {{{1
+fu! s:notifyChange()
+	let l:cb = s:getextvar('change')
+	if l:cb != -1
+		cal eval(l:cb)
+	en
+endf
+
+fu! ctrlp#set(lines)
+	let g:ctrlp_lines = a:lines
```

Comment:
Do you think this should return copy? (deepcopy)

Reply:
Not sure about this. Actually I don't want to explicitly do this here since I want the caller to optimize as much as possible first. Any reason why were you thinking of deep copy?